receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -77,6 +84,40 @@ func newMetricFamily(metricName string, mc MetadataCache) MetricFamily {
 	}
 }
 
+// Define manualy the metadata of prometheus scrapper internal metrics
+func defineInternalMetric(metricName string, metadata scrape.MetricMetadata) scrape.MetricMetadata {
+	if metadata.Metric != "" && metadata.Type != "" && metadata.Unit != "" && metadata.Help != "" {
+		//b.logger.Debug("Internal metric seems already fully defined")
+		return metadata
+	}
+	metadata.Metric = metricName
+
+	switch metricName {
+	case scrapeUpMetricName:
+		metadata.Unit = "bool"
```

Comment:
Units appear to be specific to OpenMetrics, rather than prometheus text format.  See [prometheus/prometheus/pkg/textparse/promparse.go#L195](https://github.com/prometheus/prometheus/blob/609ba54b8f7570ba23479343eec8ba3039031322/pkg/textparse/promparse.go#L195).  For OpenMetrics, here is the documentation for units: [OpenObservability/OpenMetrics/specification/OpenMetrics.md#units-and-base-units](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#units-and-base-units).  The units we should stick to include (from the OpenMetrics link): `seconds, bytes, joules, grams, meters, ratios, volts, amperes, and celsius`.  The "up" metric probably shouldn't have units in that case.